pkg/kubelet/container/interface.go

```diff
@@ -0,0 +1,408 @@
+	// of the created container.
+	Create(config *ContainerConfig, sandboxConfig *PodSandboxConfig, sandboxID PodSandboxID) (RawContainerID, error)
+	// Start starts a created container.
+	Start(id RawContainerID) error
+	// Stop stops a running container with a grace period (i.e., timeout).
+	Stop(id RawContainerID, timeout int) error
+	// Remove removes the container.
+	Remove(id RawContainerID) error
+	// List lists the existing containers that match the ContainerFilter.
+	// The returned list should only include containers previously created
+	// by this ContainerManager.
+	List(filter ContainerFilter) ([]ContainerListItem, error)
+	// Status returns the status of the container.
+	Status(id RawContainerID) (RawContainerStatus, error)
+	// Exec executes a command in the container.
+	Exec(id RawContainerID, cmd []string, streamOpts StreamOptions) error
+}
+
+// ContainerListItem provides the runtime information for a container returned
+// by List().
+type ContainerListItem struct {
+	// The ID of the container, used by the container runtime to identify
+	// a container.
+	ID ContainerID
+	// The name of the container, which should be the same as specified by
+	// api.Container.
+	Name string
+	// Reference to the image in use. For most runtimes, this should be an
+	// image ID.
+	ImageRef string
+	// State is the state of the container.
+	State ContainerState
+	// Labels are key value pairs that may be used to scope and select individual resources.
+	Labels Labels
+}
+
+type ContainerConfig struct {
+	// Name of the container. The string should conform to [a-zA-Z0-9_-]+.
+	Name string
+	// Image to use.
+	Image ImageSpec
+	// Command to execute (i.e., entrypoint for docker)
+	Command []string
+	// Args for the Command (i.e., command for docker)
+	Args []string
+	// Current working directory of the command.
+	WorkingDir string
+	// List of environment variable to set in the container
+	Env []KeyValue
+	// Mounts specifies mounts for the container
+	Mounts []Mount
+	// Labels are key value pairs that may be used to scope and select individual resources.
+	Labels Labels
+	// Annotations is an unstructured key value map that may be set by external
+	// tools to store and retrieve arbitrary metadata.
+	Annotations map[string]string
+	// Privileged runs the container in the privileged mode.
+	Privileged bool
+	// ReadOnlyRootFS sets the root filesystem of the container to be
+	// read-only.
+	ReadOnlyRootFS bool
+	// Path relative to PodSandboxConfig.LogDirectory for container to store
+	// the log (STDOUT and STDERR) on the host.
+	// E.g.,
+	//     PodSandboxConfig.LogDirectory = `/var/log/pods/<podUID>/`
+	//     ContainerConfig.LogPath = `containerName_Instance#.log`
+	//
+	// WARNING: Log managment and how kubelet should interface with the
+	// container logs are under active discussion in
+	// https://issues.k8s.io/24677. There *may* be future change of direction
+	// for logging as the discussion carries on.
+	LogPath string
```

Comment:
@philips That refers to a random uuid of (or maps to) the container, which is used to separate the current container life from the previous ones (e.g. in a crashloop), so logs don't get overridden.